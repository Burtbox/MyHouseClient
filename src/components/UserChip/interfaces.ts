import { IComponentProps } from '../../interfaces/componentInterfaces';
import { IUserObject } from '../Users/usersInterfaces';
    
export interface IUserChipProps extends IComponentProps {
    user: IUserObject;
    styles: any;
}

export interface IUserChipState {}
