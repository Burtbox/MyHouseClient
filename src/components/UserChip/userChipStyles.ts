
import { createStyles, Theme } from '@material-ui/core/styles';

const userChipStyles = (theme: Theme) => createStyles({
    chip: {
        display: 'inline-flex',
        marginRight: '4px',
    },
});

export default userChipStyles;
