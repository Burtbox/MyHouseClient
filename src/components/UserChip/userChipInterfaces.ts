import { WithStyles } from '@material-ui/core';
import { IUser } from '../Users/usersInterfaces';
import userChipStyles from './userChipStyles';

export interface IUserChipProps extends WithStyles<typeof userChipStyles> {
    user: IUser;
}
