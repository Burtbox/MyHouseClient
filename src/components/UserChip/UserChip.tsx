import Avatar from '@material-ui/core/Avatar';
import Chip from '@material-ui/core/Chip';
import * as React from 'react';
import styles from '../../styles';
import { IUserChipProps } from './userChipInterfaces';

const UserChip: React.StatelessComponent<IUserChipProps> = (props) => {
    return (
      <Chip key={'Chip_' + props.user.userId} style={styles.chip}>
        <Avatar key={'Avatar_' + props.user.userId}>
          {props.user.displayName ? props.user.displayName.charAt(0).toUpperCase() : ''}
        </Avatar>
        {props.user.displayName}
      </Chip>
    );
};

export default UserChip;
