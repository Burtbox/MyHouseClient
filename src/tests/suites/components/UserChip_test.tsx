import { shallow } from 'enzyme';
import * as React from 'react';
import UserChip from '../../../components/UserChip';
import { IUserChipProps } from '../../../components/UserChip/userChipInterfaces';

describe('User Chip test suite', () => {
    test('Render user chip', () => {
        const props: Readonly<IUserChipProps> = {
            user: {
                userId: 'unitTestUserID',
                token: 'tooken',
                email: 'unitTestEmail',
                displayName: 'unitTestDisplayName',
            },
            classes: {
                chip: 'chip',
            },
        };
        const UserChipScreen = shallow(
            <UserChip {...props} />,
        );
        expect(UserChipScreen).toMatchSnapshot();
    });
});
