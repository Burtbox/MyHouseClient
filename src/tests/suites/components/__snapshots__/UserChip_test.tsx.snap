// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Chip test suite Render user chip 1`] = `
<Chip
  containerElement="div"
  key="Chip_unitTestUserID"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyboardFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "display": "inline-flex",
      "marginRight": "4px",
    }
  }
>
  <Avatar
    key="Avatar_unitTestUserID"
    size={40}
  >
    U
  </Avatar>
  unitTestDisplayName
</Chip>
`;
